File: examples/messy-project/app/src/api/post.ts
import { request } from './client'

export interface Post {
  id: number
  title: string
  content: string
  authorId: number
  likeCount: number
  createdAt: string
}

export function listPosts(page = 1, pageSize = 20) {
  return request<{ list: Post[]; total: number }>(`/posts?page=${page}&page_size=${pageSize}`)
}

export function createPost(input: { title: string; content: string }) {
  return request<Post>('/posts', {
    method: 'POST',
    body: JSON.stringify(input),
  })
}

File: examples/messy-project/backend/internal/model/post.go
package model

import "time"

type Post struct {
	ID        int64     `json:"id"`
	Title     string    `json:"title"`
	Content   string    `json:"content"`
	AuthorID  int64     `json:"author_id"`
	LikeCount int64     `json:"like_count"`
	CreatedAt time.Time `json:"created_at"`
}
